classes/04class/aws_iam_storage/README.md

```diff
@@ -0,0 +1,540 @@
+
+## Main Use Cases
+There are four common usage patterns for Amazon S3:
+
+1. Store and distribute **static web content and media**. This content can be delivered directly from Amazon S3 because each object in S3 has a unique HTTP URL.
+   
+2. Host entire **static websites**. Amazon S3 provides a low-cost, highly available, and highly scalable solution, including storage for static HTML files, images, videos, and client-side scripts in formats such as JavaScript.
+   
+3. **Data store** for computation and large-scale analytics, such as financial transaction analysis and clickstream analytics. You can access your data from multiple computing nodes concurrently without being constrained by a single connection.
+   
+4. **Backup and archiving** of critical data.
+
+## Amazon S3 Concepts
+
+### Buckets
+A bucket is a container for objects stored in Amazon S3. 
+
+*Every* object is contained in a bucket. 
+
+For example, if the object named `photos/puppy.jpg` is stored in the `johnsmith` bucket in the US West (Oregon) Region, then it is addressable using the URL https://johnsmith.s3.us-west-2.amazonaws.com/photos/puppy.jpg.
+
+#### Important stuff
+- Bucket names must be globally unique in AWS.
+- Amazon S3 has no physical hierarchy; it only has buckets that contain objects in a [flat file structure](https://docs.amazonaws.cn/en_us/AmazonS3/latest/dev/walkthrough1.html).
+
+### Objects
+
+Objects are the fundamental entities stored in Amazon S3. 
+
+Objects consist of object *data* and *metadata*.
+
+The metadata is a set of name-value pairs that describe the object. These include some default metadata, such as the date last modified, and standard HTTP metadata, such as `Content-Type`. You can also specify custom metadata at the time the object is stored.
+
+#### Important stuff
+- Individual Amazon S3 objects can range in size from a minimum of **0 bytes** to a maximum of **5 terabytes** (to go over 5 gigabytes for an object, you need to use [multi-part upload](https://docs.aws.amazon.com/AmazonS3/latest/dev/mpuoverview.html)).
+- The largest object that can be *uploaded in a single PUT* is **5 gigabytes**. 
+- Amazon S3 offers eventual consistency for overwrite PUTS and DELETES in all Regions (the object is ***not*** deleted or update synchronously, i.e., a subsequent read might return the old data or the updated data, but it never returns corrupted or partial data.).
+
+
+### Keys
+A key is the unique identifier for an object within a bucket.
+
+Every object in a bucket has exactly one key. 
+
+The combination of a `bucket, key, and version ID`uniquely identify each object. 
+
+ For example, in the URL `http://doc.s3.amazonaws.com/2006-03-01/AmazonS3.wsdl`:
+ - `doc` is the name of the bucket
+ - `2006-03-01/AmazonS3.wsdl` is the key.
+
+> **Prefix**
+> 
+> The prefix value is similar to a directory name that enables you to store similar data under the same directory in a bucket.
```

Comment:
`The prefix value is similar to a directory name that enables you to store similar data under the same directory in a bucket.`
This indicates that there is directories in Buckets, which is not correct. Maybe:
`The prefix value is similar to a directory in a traditional filesystem, and enables you to store similar data in an organised way in a bucket.`